include/eve/detail/function/simd/arm/neon/reverse.hpp

```diff
@@ -0,0 +1,31 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/abi.hpp>
+#include <eve/detail/function/swap_adjacent_groups.hpp>
+
+namespace eve::detail
+{
+  template<typename T, typename N, arm_abi ABI>
+  EVE_FORCEINLINE wide<T,N,ABI> reverse_( EVE_SUPPORTS(neon128_), wide<T,N,ABI> v) noexcept
+  {
+    constexpr bool one_instruction_basic_swizzle = (current_api >= asimd) || std::same_as<ABI, arm_64_>;
+
+    // two reverse instructions seem better than one table lookup.
+    if constexpr ( N() >= 8 && one_instruction_basic_swizzle) return reverse_(EVE_RETARGET(cpu_), v);
+    else
+    {
+      if constexpr ( N() >= 16 ) v = eve::swap_adjacent_groups(v, lane<8>);
```

Comment:
can you clarify the algorithm here ? Why is it enough ?

Reply:
Well, you can't really reverse 16 chars with one instruction.
But you can:
Swap two 64 bit groups => reverse each halve (so - swap 32 bit groups => reverse each halve etc)

Reply:
ok